eta/core/video.py

```diff
@@ -553,23 +554,30 @@ class VideoLabels(Serializable):
     Note that any falsey fields of this class will be omitted during
     serialization.
 
-    Note that when VideoLabels objects are serialized, the keys of the `frames`
-    dict will be converted to strings, because all JSON object keys _must_ be
+    Note that when VideoLabels objects are serialized, any integer keys or
+    fields will be converted to strings and then back to integers upon
+    deserialization.  For example, the keys of the `frames` dict will be
+    converted to strings, because all JSON object keys _must_ be
     strings. The `from_dict` method of this class handles converting the keys
     back to integers when VideoLabels instances are loaded.
 
+    Frames are the primary representation-mechanism for VideoLabels.  So, when
+    using VideoLabels in a dict-like fashion, frames are retrieved, not
+    attributes and not events.
+
     Attributes:
         filename: the filename of the video
         metadata: a VideoMetadata describing metadata about the video
         attrs: an AttributeContainer containing video attributes
         frames: a dictionary mapping frame number strings to VideoFrameLabels
             instances
+        events: a DetectedEventContainer containing events
```

Comment:
A common pattern for working with `VideoLabels` is:

```py
video_labels = etav.VideoLabels(...)

for frame_number in VideoLabels:
    frame_labels = video_labels[frame_number]  # eta.core.video.VideoFrameLabels
    # do stuff with the labels for the frame
```

This will not contain any attributes for events that coincide with a given frame.

Now that VideoLabels contains semantic things, we need to decide how to deal with frame-based iteration. Do we add support for "rendering" events for a given frame, which are stored in `VideoFrameLabels` "on-demand"?

We need a way to iterate over the frames of a VideoLabels that contains _everything_ for that frame. Player51 needs that, for example.

Reply:
My view, again, is that this not any more "semantic" than what there, but I think you are using that word for something different than I do.  If VideoLabels "needs" to remain only frame-based, then the proper thing to do is move events  to frames like objects.  However, I'll push back on this assumption "We need a way to iterate over the frames of a..."  Why?  Many representations require the rendered to ingest, visualize.  This is just a tacit assumption that has been made.